apps/wallet/src/ui/app/shared/faucet/message-info/index.tsx

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { motion, AnimatePresence } from 'framer-motion';
+import { useIntl } from 'react-intl';
+
+import Alert from '_components/alert';
+import { useAppSelector, useFormatCoin } from '_hooks';
+import { GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+
+import type { AlertProps } from '_components/alert';
+import type { IntlShape } from 'react-intl';
+
+const ONE_MINUTE = 60;
+const ONE_HOUR = ONE_MINUTE * 60;
+const ONE_DAY = ONE_HOUR * 24;
+
+function formatError(
+    intl: IntlShape,
+    status: number,
+    statusTxt: string | undefined,
+    retryAfter: number | undefined
+) {
+    if (status === 429) {
+        let retryTxt = 'later';
+        if (retryAfter) {
+            let unit = 'second';
+            let value = retryAfter;
+            if (retryAfter > ONE_DAY) {
+                unit = 'day';
+                value = Math.floor(retryAfter / ONE_DAY);
+            } else if (retryAfter > ONE_HOUR) {
+                unit = 'hour';
+                value = Math.floor(retryAfter / ONE_HOUR);
+            } else if (retryAfter > ONE_MINUTE) {
+                unit = 'minute';
+                value = Math.floor(retryAfter / ONE_MINUTE);
+            }
+            retryTxt = intl.formatNumber(value, {
+                style: 'unit',
+                unit,
+                unitDisplay: 'long',
+            });
+        }
+        return `Request limit reached, please try again after ${retryTxt}.`;
+    }
+    return `Gas request failed${statusTxt ? `, ${statusTxt}` : ''}.`;
+}
+
+export type FaucetMessageInfoProps = {
+    className?: string;
+};
+
+function FaucetMessageInfo({ className }: FaucetMessageInfoProps) {
```

Comment:
I think this is something that we should generalize. I used `react-hot-toast` on the explorer side that also probably could be incorporated here. We can make this a follow-up task for another sprint though.

Reply:
Yeah thats how I started to implement this but the positioning of the message/toast made it a bit difficult - so I did it this way to make sure I would have it ready for the release. But yeah, we need a follow-up task to make a generic toast/notifications system.